api/v2beta1/condition_types.go

```diff
@@ -16,6 +16,10 @@ limitations under the License.
 
 package v2beta1
 
+// ChartFinalizer is set on a HelmRelease when a HelmChart object is created
+// for it, and removed when this object has been deleted.
+const ChartFinalizer = "chart.finalizers.fluxcd.io"
```

Comment:
What's the role of this finalizer?

Reply:
It is to prevent the HelmRelease from being removed while the HelmChart has not been cleaned up yet by the separate reocnciler.

Reply:
So if I do `flux delete hr` the HR will not be deleted not the Helm release uninstalled until SC GCs the HelmChart?

Reply:
The Helm release will be uninstalled but the release will not be removed until the HelmChart has successfully been removed.

Reply:
Note this is not different from the current logic, which does this as well https://github.com/fluxcd/helm-controller/blob/main/controllers/helmrelease_controller.go#L649-L652.

Reply:
> until the HelmChart has successfully been removed

Does it want for it to be removed from etcd or juat marked for deleting? Does this means HRs will hang indefinitely if SC fails to finalize the chart?

Reply:
Ok it just marks the chart for deletion instead of blocking. All good.